File: other/ciklus1.mjs
console.log("- 1. feladat");
for (let i = 10; i --> 0; ) if ((10 - i) % 2 === 0) console.log(10 - i);

console.log(
    new Array(10)
        .fill(1)
        .map((_, i) => i + 1)
        .filter((x) => !(x % 2))
        .join(", ")
);

console.log("\n- 2. feladat");
for (let i = 11; i --> 1; ) console.log(i);

console.log(
    new Array(10)
        .fill(1)
        .map((_, i) => 10 - i)
        .join(", ")
);

console.log("\n- 3. feladat");
for (let i = 10; i --> 0; ) if (i % 2) console.log(i);

console.log(
    new Array(10)
        .fill(1)
        .map((_, i) => 10 - i)
        .filter((x) => x % 2)
        .join(", ")
);

const readline = (out) =>
    new Promise((resolve) => {
        let input = "";
        const listener = (data) => {
            input += data.toString();
            if (!input.endsWith("\n")) return;
            process.stdin.off("data", listener);
            resolve(input.slice(0, -1));
        };
        process.stdin.on("data", listener);
        process.stdout.write(out);
    });

console.log("\n- 4. feladat");

const text = await readline("Mi legyen a szöveg? ");
const n = await readline("Hányszor írja ki a szöveget? ").then(Number);
for (let i = n; i --> 0; ) console.log(text);

console.log("\n- 5. feladat");
while (await readline("Egy páros szám: ").then(Number) % 2);

console.log("\n- 6. feladat");
const nums = new Array(20).fill(1).map(() => Math.ceil(Math.random() * 12));
const three = nums.filter(x => !(x % 3));
console.log(three.join(", "));
console.log(`Összesen ${three.length} darab hárommal osztható szám lett generálva.`);

process.exit(0);

File: other/ciklus1.py
print("- 1. feladat")
for n in range(1, 11):
    if n % 2 == 0: print(n)

print(", ".join([str(n) for n in range(2, 11, 2)]))

print("\n- 2. feladat")
for n in range(10, 0, -1):
    print(n)

print(", ".join([str(n) for n in range(10, 0, -1)]))

print("\n- 3. feladat")
for n in range(9, 0, -2):
    print(n)

print(", ".join([str(n) for n in range(9, 0, -2)]))

print("\n- 4. feladat")
text = input("Mi legyen a szöveg? ")
n = int(input("Hányszor írja ki a szöveget? "))
for _ in range(n):
    print(text)

print("\n- 5. feladat")
while int(input("Egy páros szám: ")) % 2 == 1: pass

print("\n- 6. feladat")
from random import randint as ri
nums = [ri(1, 12) for _ in range(20)]
three = list(filter(lambda x: x % 3 == 0, nums))
print(", ".join([str(x) for x in three]))
print(f"Összesen {len(three)} darab hárommal osztható szám lett generálva.")
